lab/src/Microsoft.ServiceModel.Syndication/src/Atom10FeedFormatter.cs

```diff
@@ -763,155 +868,160 @@ private SyndicationContent ReadContentFrom(XmlReader reader, SyndicationItem ite
-                                }
-                                string val = reader.Value;
-                                if (!TryParseAttribute(name, ns, val, result, this.Version))
-                                {
-                                    if (_preserveAttributeExtensions)
-                                    {
-                                        result.AttributeExtensions.Add(new XmlQualifiedName(reader.LocalName, reader.NamespaceURI), reader.Value);
-                                    }
-                                    else
-                                    {
-                                        TraceSyndicationElementIgnoredOnRead(reader);
-                                    }
+                                    result.AttributeExtensions.Add(new XmlQualifiedName(reader.LocalName, reader.NamespaceURI), await reader.GetValueAsync());
                                 }
                             }
                         }
                     }
-                    reader.ReadStartElement();
                 }
+                await reader.ReadStartElementAsync();
+            }
 
-                XmlBuffer buffer = null;
-                XmlDictionaryWriter extWriter = null;
-                bool areAllItemsRead = true;
-                bool readItemsAtLeastOnce = false;
-
-                if (!elementIsEmpty)
+            XmlBuffer buffer = null;
+            XmlDictionaryWriter extWriter = null;
+            bool areAllItemsRead = true;
+            bool readItemsAtLeastOnce = false;
+            
+            if (!elementIsEmpty)
+            {
+                try
                 {
-                    try
+                    while (await reader.IsStartElementAsync())
                     {
-                        while (reader.IsStartElement())
+                        if (await TryParseFeedElementFromAsync(reader, result))  
                         {
-                            if (TryParseFeedElementFrom(reader, result))
+                            // nothing, we parsed something, great
+                        }
+                        else if (await reader.IsStartElementAsync(Atom10Constants.EntryTag, Atom10Constants.Atom10Namespace) && !isSourceFeed)
+                        {
+                            if (readItemsAtLeastOnce)
                             {
-                                // nothing, we parsed something, great
+                                //throw new InvalidOperationException(String.Format(SR.FeedHasNonContiguousItems, this.GetType().ToString()));
+                                //Log we have disjoint items
                             }
-                            else if (reader.IsStartElement(Atom10Constants.EntryTag, Atom10Constants.Atom10Namespace) && !isSourceFeed)
+
+
+                            while (await reader.IsStartElementAsync(Atom10Constants.EntryTag, Atom10Constants.Atom10Namespace))
                             {
```

Comment:
You've removed an extensibility point here by not using ReadItemsAsync which is a protected virtual method. Keep ReadItemsAsync being called. Instead of having the out parameter of areAllItemsRead, we can have the contract be that they don't complete the returned Task until all items have been read.